devcareer/code-jammers-backend
# #174566162 Build out endpoint for user sign in

#### What does this PR do

This PR enable a user to sign in

#### Description of tasks to be completed?

-    Install relevant dependencies
-    Write program to sign in a user
-    Test program to ensure user is signed in successfully

#### How should this be manually tested/checked?

-    Clone the PR
-    Cd into the branch ft-user-sign-in-174566162
-    Set up you enviroment varaibles(check the .envsample)
-    Install the dependencies using `npm install`
-    Set up migration run `npm run migrate`
-    Start the server by running `npm run start-dev`
-    Test the sign up post route, using Postman with `http://localhost:3000/api/v1/users/signup` as url.
-    A user must signed up with a username, email and a password
-    confirm user account was created
-    Test the sign in post route, using Postman with `http://localhost:3000/api/v1/users/signin`
-    A user must  sign in with an email and a password
-    confirm user was sign in successfully
-    Also test for validations by inputing wrong details for username, email, password

## What are the relevant pivotal tracker stories? ##
[#174566162](https://www.pivotaltracker.com/story/show/174566162)

## Architectural flowchart ##
[Create User sign in google docs](https://docs.google.com/document/d/1migRMzrITgl8BiuXlho-huPA_GYlTARh2g7NMwJqOUA/edit)
